### PR TITLE
New Zed Root Directory

### PR DESCRIPTION
**Pull request by jameskerr**
1. Use the zed lake class in the zealot folder.
2. Use the zdeps binary for zed
3. Change the root directory for the lake to "userdata/lake".

This allows the app to start up without error.

Fixes: #2156

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -88,11 +88,11 @@
...
     "husky": "^7.0.4",
-    "jest": "^27.4.5",
-    "jest-circus": "^27.4.6",
-    "jest-environment-node": "^27.4.6",
+    "jest": "^28.0.0-alpha.0",
+    "jest-circus": "^28.0.0-alpha.0",
+    "jest-environment-node": "^28.0.0-alpha.0",
     "jest-playwright-preset": "^1.7.0",
-    "jest-runner": "^27.4.6",
+    "jest-runner": "^28.0.0-alpha.0",
```
Comment (jameskerr) [inline package.json:95]: I had to upgrade jest to this new version to import "@brimdata/zealot/node" in the tests. The package.json "exports" field is new and just getting support.

File: packages/zealot/package.json [hunk 2]
```diff
@@ -12,10 +12,26 @@
...
+  "typesVersions": {
+    "*": {
+      "*": [
+        "dist/cjs/index.d.ts"
+      ],
+      "node": [
+        "./dist/cjs/node.d.ts"
+      ]
+    }
+  },
```
Comment (jameskerr) [inline packages/zealot/package.json:34]: A hack to have typescript support the subfolder exports.

File: src/js/electron/main.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ require("@electron/remote/main").initialize()
 
 export const mainDefaults = () => ({
   lakePort: 9867,
-  lakeRoot: join(app.getPath("userData"), "data", "lake"),
+  lakeRoot: join(app.getPath("userData"), "lake"),
```
Comment (jameskerr) [inline src/js/electron/main.ts:32]: This is the whole point of this PR

**Review by mason-fish — APPROVED** [review]
This looks good to me pending a green build.

One question though: if we are effectively dropping all existing lakes and starting a new one with this change, should we also include a migration to reset the rest of the app state?

File: packages/zealot/src/cmd/paths.ts [hunk 4]
```diff
@@ -3,6 +3,13 @@ import os from "os"
...
+  let dir = "../../../../node_modules/zed/dist"
+  if (isBuilt()) dir = "../" + dir
+  return path.join(__dirname, dir, name)
+}
 
-  return path.join(__dirname, "../../../../node_modules/zed/dist", name)
+function isBuilt() {
+  const buildDir = path.normalize("dist/cjs/cmd")
+  const regex = new RegExp(`${buildDir}$`)
+  return regex.test(__dirname)
```
Comment (nwt) [inline packages/zealot/src/cmd/paths.ts:14]:
```suggestion
  const buildDir = path.normalize("dist/cjs/cmd")
  return __dirname.endsWith(buildDir)
```

**Comment by jameskerr** [comment]
@mason-fish good point. The app state will remain unchanged, but the data for the zed lake will be new. In the release notes we will provide a migration script for people to run that will bring the data from the old lake into the new one.

File: test/playwright/playwright.config.js [hunk 5]
```diff
@@ -1,4 +1,6 @@
 module.exports = {
+  globalSetup: "./setup/global.ts",
+  testDir: "./tests",
   browsers: [
```
Comment (mason-fish) [inline test/playwright/playwright.config.js:4]: I don't think this browsers key is a playwright config options and can now be removed. It was only used with the jest setup to rename the test browser in each test's output (it's default output did not indicate electron at all despite the fact that it was driving our electron app)

**Review by mason-fish — COMMENTED** [review]
@jameskerr The new test runner looks good! Part of why I didn't use it originally was that it wasn't resolving zealot properly, so this actually makes perfect sense to do now. And with this in place we can also begin to use their very nice [trace-on-failure](https://playwright.dev/docs/test-configuration#record-test-trace) features for debugging flakiness 👌

File: test/playwright/playwright.config.js (hunk 5 above)
Comment (jameskerr) [reply test/playwright/playwright.config.js:4]: Awesome!

**Pull request merged**